Sources/CommandFramework/AbstractCommand.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import SourceParsingFramework
+import Utility
+
+/// The base class of all commands that perform a set of logic common
+/// to all commands.
+open class AbstractCommand: Command {
+    /// The name used to check if this command should be executed.
+    public let name: String
+
+    /// The default value for waiting timeout in seconds.
+    public let defaultTimeout = 30.0
```

Comment:
nit: I would add a `: TimeInterval` here so that the type system expresses that the unit is seconds in addition to the comment.

Reply:
Sounds good.